pkg/codegen/python/gen.go

```diff
@@ -1205,18 +1223,24 @@ func (mod *modContext) genResource(res *schema.Resource) (string, error) {
 		fmt.Fprintf(w,
 			"        opts = pulumi.ResourceOptions.merge(opts, pulumi.ResourceOptions(id=id))\n")
 		fmt.Fprintf(w, "\n")
-		fmt.Fprintf(w, "        __props__ = dict()\n\n")
+		if hasStateInputs {
+			fmt.Fprintf(w, "        __props__ = _%[1]sState.__new__(_%[1]sState)\n\n", name)
+		} else {
+			// If we don't have any state inputs, we'll just instantiate the `<Resource>Args` class,
+			// to opt-in to the improved translation behavior.
+			fmt.Fprintf(w, "        __props__ = %[1]sArgs.__new__(%[1]sArgs)\n\n", name)
+		}
 
 		stateInputs := codegen.NewStringSet()
 		if res.StateInputs != nil {
 			for _, prop := range res.StateInputs.Properties {
 				stateInputs.Add(prop.Name)
-				fmt.Fprintf(w, "        __props__[%[1]q] = %[1]s\n", PyName(prop.Name))
+				fmt.Fprintf(w, "        __props__.__dict__['%[1]s'] = %[1]s\n", PyName(prop.Name))
```

Comment:
Oh interesting, you actually changed the `%q` to `'%s'` - are you trying to standardize to single quotes?
```suggestion
				fmt.Fprintf(w, "        __props__.__dict__[%[1]q] = %[1]s\n", PyName(prop.Name))
```

Reply:
I was just changing `get` to match `__init__` so that they were both setting `__props__` using the same quote style. I've now changed both to use double quotes.